### PR TITLE
introduce respectWhitespace parse option

### PR DESCRIPTION
**Pull request by adrai**
should help to fix: https://github.com/i18next/react-i18next/pull/1283
originally from here: https://github.com/rayd/html-parse-stringify2/blob/master/lib/parse.js#L14

### CODE REVIEW

**Comment by adrai** [comment]
@HenrikJoreteg any reaction from your side would help a lot here:
https://github.com/i18next/react-i18next/issues/1275#issuecomment-817272933
https://github.com/i18next/react-i18next/pull/1283#issuecomment-817273004

File: src/parse.js [hunk 1]
```diff
@@ -102,9 +102,17 @@ export default function parse(html, options) {
...
+        // if a node is nothing but whitespace, collapse it as the spec states:
+        // https://www.w3.org/TR/html4/struct/text.html#h-9.1
+        if (/^\s*$/.test(content)) {
+          content = ' ';
+        }
+        // don't add whitespace-only text nodes if they would be trailing text nodes
+        // or if they would be leading whitespace-only text nodes:
+        //  * end > -1 indicates this is not a trailing text node
+        //  * leading node is when level is -1 and parent has length 0
+        if ((options.respectWhitespace && end > -1 && level + parent.length >= 0) || content !== ' ') {
```
Comment (HenrikJoreteg) [inline src/parse.js:115]: I'm trying to figure out if there's a scenario where you would not want `respectWhitespace` to be true. 

I'd rather just change behavior than add an option, unless we can think of a good reason why anyone would ever want it to be false.

**Review by HenrikJoreteg — COMMENTED** [review]
Thanks for this @adrai. I just have one comment, really.

**Comment by HenrikJoreteg** [comment]
I'm tempted to just do a major version bump and change the behavior to always preserve it, what do you think?

**Comment by adrai** [comment]
I was thinking the same when creating the PR, but did not want to break the existing behaviour.
If you want to create a major and always respect the whitespaces it's even better ;-)

File: src/parse.js (hunk 1 above)
Comment (adrai) [reply src/parse.js:115]: I was thinking the same when creating the PR, but did not want to break the existing behaviour.
If you want to create a major and always respect the whitespaces it's even better ;-)

**Comment by adrai** [comment]
So let's close this PR

**Pull request closed without merging**